osquery/events/darwin/openbsm.cpp

```diff
@@ -47,33 +50,32 @@ Status OpenBSMEventPublisher::configureAuditPipe() {
   }
 
   au_mask_t pr_flags = {0, 0};
-  std::vector<std::string> ev_classes;
+  std::set<std::string> ev_classes;
 
   if (true == FLAGS_audit_allow_process_events) {
     // capture process events
-    ev_classes.push_back("pc");
+    ev_classes.insert("pc");
   }
 
   if (true == FLAGS_audit_allow_sockets) {
     // capture network events
-    ev_classes.push_back("nt");
+    ev_classes.insert("nt");
   }
 
   if (true == FLAGS_audit_allow_user_events) {
     // capture user (login, autherization etc...) events
-    ev_classes.push_back("lo");
-    ev_classes.push_back("aa");
-    ev_classes.push_back("ad");
+    ev_classes.insert("lo");
+    ev_classes.insert("aa");
   }
 
   if (true == FLAGS_audit_allow_fim_events) {
     // capture file events
-    ev_classes.push_back("fc");
-    ev_classes.push_back("fd");
-    ev_classes.push_back("fw");
-    ev_classes.push_back("fr");
-    ev_classes.push_back("fa");
-    ev_classes.push_back("fm");
+    ev_classes.insert("fc");
+    ev_classes.insert("fd");
+    ev_classes.insert("fw");
+    ev_classes.insert("fr");
+    ev_classes.insert("fa");
+    ev_classes.insert("fm");
   }
 
   struct au_class_ent* ace = nullptr;
```

Comment:
How about?
  while ((ace = getauclassent()) != nullptr) { 
    auto it = ev_classes.find(ace->ac_name);
    if (it != ev_classes.end()) {
      ADD_TO_MASK(&pr_flags, ace->ac_class, AU_PRS_BOTH);	
    }
  }

Reply:
Since this only runs once I don't think we need to be super optimized.

Reply:
good point, then why changing from std::vector to std::set?

Reply:
I guess because we started using `set` for `event_ids` and it's the more appropriate container for this code here too, so I figured why not.

Reply:
Which property of set are we using here? We are iterating all the elements here for which vector or array is sufficient, I believe.